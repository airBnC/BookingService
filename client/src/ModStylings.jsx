import styled from 'styled-components';
import logo from './images/arrowLogo.png';
import arrow from './images/arrow.png';

export const Holder = styled.div`
display: flex;
flex-direction: column;
width: inherit;
padding-top: 10px;
`;

export const InputHolder = styled.div`
  display: flex;
  flex-direction: row;
  justify-content: space-between;
  border: 1px solid rgb(172, 172, 172);
  padding: 5px;
  padding-left: 15px;
  padding-right: 10px;
  &:hover{
    cursor: pointer;
  }
`;

export const Title = styled.div`
  font-size: 12px;
  font-weight: bold;
  margin-bottom: 3px;
`;

export const Logo = styled.div`
  content: url(${logo});
  height: 17px;
  width: 25px;
  transform: rotate(${props => (props.clicked ? '0turn' : '0.5turn')});
`;

export const Arrow = styled.div`
  content: url(${arrow});
  height: 17px;
`;

export const DropdownHolder = styled.div`
  position: absolute;
  z-index: 2;
  display: flex;
  flex-direction: column;
  width: 286px;
  border: 1px solid rgb(172, 172, 172);
  border-radius: 1%;
  padding: 20px;
  background-color: white;
  user-select: none;
`;

export const Close = styled.button`
  margin-top: 8px;
  border: none;
  align-self: flex-end;
  font-size: 14px;
  color: #007D8C;
  outline: none;
  &:hover{
    text-decoration: underline;
  }
  &:active{
    color: #FD5C63;
  }
`;
