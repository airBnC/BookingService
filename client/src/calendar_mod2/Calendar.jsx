import React from 'react';
import PropTypes from 'prop-types';
import moment from 'moment';
import { Table, Tr, Button } from './CalendarStyling';
import { createDaysArr, findOccupiedDatesInMonth } from './CalendarLogic';

export default function Calendar(props) {
  const onDateClick = (day) => {
    if (props.clicked === 1) {
      if (props.endDate) {
        if (day > props.endDate || day + props.minStay > props.endDate) {
          props.clearDates();
        }
      }
      props.dateClick('startDate', day);
      props.calendarChange('checkOutClicked');
    } else if (props.clicked === -1) {
      props.dateClick('endDate', day);
      props.calendarChange();
    }
  };

  const daysArr = createDaysArr({
    year: props.year,
    month: props.month,
  });

  let occupiedDates;
  if (props.clicked === 1) {
    occupiedDates = findOccupiedDatesInMonth({
      month: props.month,
      year: props.year,
      dates: props.dates,
    });
  } else if (props.clicked === -1) {
    occupiedDates = findOccupiedDatesInMonth({
      startDate: props.startDate,
      month: props.month,
      year: props.year,
      minStay: props.minStay,
      dates: props.dates,
    });
  }

  const calendar = daysArr.map((day, i) => {
    let isOccupied = false;
    if (occupiedDates.includes(day) || day === '') {
      isOccupied = true;
    }
    return (
      <Button
        key={i}
        day={day}
        date={isOccupied}
        onClick={() => onDateClick(day)}
        minStay={props.minStay}
        startDate={props.startDate}
        endDate={props.endDate}
        disabled={isOccupied || (props.startDate === day)}
      >
        {day}
      </Button>
    );
  });

  const weekDayNames = moment.weekdaysMin().map(day => (
    <Button week key={day} disabled> {day} </Button>
  ));

  return (
    <Table>
      <thead>
        <Tr>
          <td>
            {weekDayNames}
          </td>
        </Tr>
      </thead>
      <tbody>
        <Tr>
          <td>
            {calendar}
          </td>
        </Tr>
      </tbody>
    </Table>
  );
}

Calendar.propTypes = {
  dates: PropTypes.arrayOf(PropTypes.string),
  minStay: PropTypes.number,
  startDate: PropTypes.number,
  endDate: PropTypes.number,
  dateClick: PropTypes.func,
  clearDates: PropTypes.func,
  calendarChange: PropTypes.func,
  // arrowClick: PropTypes.func,
  clicked: PropTypes.number,
  // currDate: momentPropTypes.momentObj,
  month: PropTypes.string,
  year: PropTypes.number,
};

Calendar.defaultProps = {
  dates: [],
  minStay: 0,
  startDate: null,
  endDate: null,
  dateClick: () => null,
  clearDates: () => null,
  calendarChange: () => null,
  // arrowClick: () => null,
  clicked: 0,
  // currDate: null,
  month: null,
  year: null,
};
