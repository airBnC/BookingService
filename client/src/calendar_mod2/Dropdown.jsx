import React from 'react';
import styled from 'styled-components';
import PropTypes from 'prop-types';
import CheckInCalendar from './CheckInCalendar';
import CheckOutCalendar from './CheckOutCalendar';
import arrow from '../images/arrow.png';

export const Holder = styled.div`
  position: absolute;
  z-index: 2;
  display: flex;
  flex-direction: column;
  width: 286px;
  border: 1px solid rgb(172, 172, 172);
  border-radius: 1%;
  padding: 20px;
  background-color: white;
`;

export const Arrow = styled.div`
  border-radius: 10%;
  content: url(${arrow});
  height: 20px;
  border: 1px solid rgb(172, 172, 172);
  border-radius: 10%;
  padding: 3px;
  transform: ${props => (props.left ? 'rotate(0.5turn)' : '')};
    &:active{
      background-color: #79CCCD;
    }
    &:hover {
      cursor: pointer;
    }
`;

export const SectionOne = styled.div`
  display: flex;
  flex-direction: row;
  justify-content: space-between;
  margin-top: 10px;
  font-size: 18px;
  font-weight: bold;
`;

export const SectionTwo = styled.div`
  display: flex;
  align-items: center;
`;

export const SectionThree = styled.div`
  font-size: 12px;
  margin-top: 12px;
  display: flex;
  flex-direction: column;
`;

export default function Dropdown(props) {
  let calendar;
  let holderArrow;
  if (props.clicked === 1) {
    calendar = (<CheckInCalendar {...props} />);
    holderArrow = (
      <svg height="10" width="250" fill="rgb(172, 172, 172)" stroke="rgb(172, 172, 172)" position="absolute">
        <path d="M 15,25 35, 25 25, 0 Z" height="0px" position="absolute" />
      </svg>);
  } else if (props.clicked === -1) {
    calendar = (<CheckOutCalendar {...props} />);
    holderArrow = (
      <svg height="10" width="250" fill="rgb(172, 172, 172)" stroke="rgb(172, 172, 172)">
        <path d="M 0,25 35, 25 25, 15 Z" height="0px" />
      </svg>);
  }
  if (props.clicked === 1 || props.clicked === -1) {
    return (
      <div>
        {holderArrow}
        <Holder>
          <SectionOne>
            <Arrow left />
            <div> {props.month} {props.year} </div>
            <Arrow />
          </SectionOne>
          <SectionTwo>
            {calendar}
          </SectionTwo>
          <SectionThree>
            <div> {props.minStay} day minimum. </div>
            <div> Updated 2 days ago </div>
          </SectionThree>
        </Holder>
      </div>
    );
  }
  return null;
}

Dropdown.propTypes = {
  clicked: PropTypes.number,
  month: PropTypes.string,
  year: PropTypes.number,
  dateClick: PropTypes.func,
  // dates: PropTypes.arrayOf( PropTypes.shape( {
  //   date: PropTypes.date,
  // } ) ),

};
Dropdown.defaultProps = {
  clicked: 0,
  month: 'June',
  year: 2018,
  dateClick: () => null,
  // dates: [],
};
