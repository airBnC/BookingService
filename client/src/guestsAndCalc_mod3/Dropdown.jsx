import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import DropdownContents from './DropdownContents';

const Holder = styled.div`
  display: flex;
  flex-direction: column;
  width: 230px;
  border: 1px solid rgb(172, 172, 172);
  padding: 10px;
  padding-top: 0px;
`;

const Details = styled.div`
  font-size: 12px;
  margin-top: 10px;
`;

export default function Dropdown( props ) {
  Dropdown.propTypes = {
    clicked: PropTypes.bool,
    maxGuests: PropTypes.number,
    adult: PropTypes.number,
    child: PropTypes.number,
    infant: PropTypes.number,
    btnClick: PropTypes.func,
  };
  Dropdown.defaultProps = {
    clicked: false,
    maxGuests: 0,
    adult: 0,
    child: 0,
    infant: 0,
    btnClick: () => null,
  };
  if ( props.clicked ) {
    console.log( 'I was clicked!' );
    return (
      <Holder>
        <DropdownContents
          adult={props.adult}
          child={props.child}
          infant={props.infant}
          btnClick={props.btnClick}
        />
        <Details> {props.maxGuests } guests maximum. Infants don't count toward the number of guests. </Details>
      </Holder>
    );
  }
  return null;
}
