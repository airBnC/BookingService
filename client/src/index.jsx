import React, { Component } from 'react';
import { render } from 'react-dom';
import styled, { injectGlobal } from 'styled-components';
import axios from 'axios';
import ModOne from './pricing_mod1/ModOne';
import ModTwo from './calendar_mod2/ModTwo';
import ModThree from './guests_mod3/ModThree';
import { setStartOrEndDate, clearDates, calendarChange } from './calendar_mod2/CalendarLogic';

injectGlobal([`
  html, body{
    height: 100%;
    width: 100%
  }
  body {
    display: flex;
    justify-content: center;
    align-items: center;
  }
`]);

export const Holder = styled.div`
  display: flex;
  justify-content: center;
  align-items: center;
  flex-direction: column;
  width: 328px;  
  height: 300px;
  border: 1px solid rgb(172, 172, 172);
  font-family: Quicksand;
  padding: 24px;
`;

export default class App extends Component {
  constructor(props) {
    super(props);
    this.state = {
      adults: 1,
      children: 0,
      infants: 0,
      totalGuests: 1,
      listingData: null,
      startDate: null,
      endDate: null,
      mod2Clicked: 'closed',
      mod3Clicked: false,
    };
    this.setStartOrEndDate = setStartOrEndDate.bind(this);
    this.clearDates = clearDates.bind(this);
    this.calendarChange = calendarChange.bind(this);
  }

  componentDidMount() {
    this.getListingData();
  }

  onGuestButtonClick = (guest, increment) => {
    const val = this.state[guest] + increment;
    const { totalGuests } = this.state;
    const total = totalGuests + increment;
    const { maxGuests } = this.state.listingData;
    // infants don't count towards the number of guests, but have a cap of 5
    if (guest === 'infants' && (val > 5 || val < 0)) {
      return;
    // there is a minimum of 1 adult guest
    } else if (guest === 'adults' && (val > maxGuests || val < 1 || total > maxGuests)) {
      return;
    } else if (guest === 'children' && (val > maxGuests || val < 0 || total > maxGuests)) {
      return;
    }
    this.setState({
      [guest]: val,
      totalGuests: guest === 'infants' ? totalGuests : total,
    });
  }

  onMod3InputHolderClick = () => {
    this.setState({
      mod3Clicked: !this.state.mod3Clicked,
    });
  }

  getListingData = () => {
    let id = window.location.pathname;
    if (id === '/') {
      id = 1;
    } else {
      id = id.replace(/\//g, '');
    }
    axios.get(`http://127.0.0.1:3002/rooms/${id}/bookingInfo/`)
      .then((response) => {
        this.setState({
          listingData: response.data,
        });
      })
      .catch(() => {
        console.log('there was an error!');
      });
  }

  render() {
    if (this.state.listingData) {
      return (
        <Holder>
          <ModOne
            price={this.state.listingData.pricePerNight}
            rating={this.state.listingData.starRating}
            numReviews={this.state.listingData.custRevNum}
          />
          <ModTwo
            dates={this.state.listingData.datesTaken}
            minStay={this.state.listingData.minStay}
            startDate={this.state.startDate}
            endDate={this.state.endDate}
            setDate={this.setStartOrEndDate}
            calendarChange={this.calendarChange}
            clicked={this.state.mod2Clicked}
            clearDates={this.clearDates}
          />
          <ModThree
            adult={this.state.adults}
            child={this.state.children}
            infant={this.state.infants}
            totalGuests={this.state.totalGuests}
            btnClick={this.onGuestButtonClick}
            maxGuests={this.state.listingData.maxGuests}
            clicked={this.state.mod3Clicked}
            close={this.onMod3InputHolderClick}
          />
          {/* <ModFour /> */}
        </Holder>
      );
    }
    return null;
  }
}

render(<App />, document.getElementById('root'));

// cleaningFee={this.state.listingData.cleaningFee}
// maxGuests={this.state.listingData.maxGuests}
// serviceFee={this.state.listingData.serviceFee}
