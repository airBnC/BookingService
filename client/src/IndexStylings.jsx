import styled, { injectGlobal } from 'styled-components';
import { abnbGrey, abnbRed } from './ModStylings';

injectGlobal([`
  html, body{
    height: 100%;
    width: 100%
  }
  body {
    display: flex;
    justify-content: center;
    align-items: center;
  }
`]);

export const Holder = styled.div`
  user-select: none;
  display: flex;
  justify-content: center;
  align-items: center;
  flex-direction: column;
  width: 328px;  
  height: ${props => ((props.startDate && props.endDate) ? '400px' : '300px')};
  border: 1px solid ${abnbGrey};
  font-family: Roboto, sans-serif;
  padding: 24px;
`;

export const Button = styled.button`
  width: inherit;
  margin-top: 20px;
  height: 40px;
  border-radius: 7px;
  outline: none;
  background-color: ${abnbRed};
  font-weight: bold;
  color: white;
`;

export const Details = styled.div`
  font-size: 12px;
  font-weight: bold;
  align-self: center;
  margin-top: 7px;
`;
