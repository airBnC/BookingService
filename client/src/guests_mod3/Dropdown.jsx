import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import DropdownContents from './DropdownContents';
import { DropdownHolder, Close } from '../ModStylings';

export const Details = styled.div`
  font-size: 12px;
  margin-top: 12px;
`;

export default function Dropdown(props) {
  if (props.clicked) {
    return (
      <DropdownHolder guest>
        <DropdownContents
          {...props}
        />
        <Details>
          {props.maxGuests } guests maximum.
          Infants donâ€™t count toward the number of guests.
        </Details>
        <Close onClick={props.close}> Close </Close>
      </DropdownHolder>
    );
  }
  return null;
}

Dropdown.propTypes = {
  clicked: PropTypes.bool,
  btnClick: PropTypes.func,
  close: PropTypes.func,
  adult: PropTypes.number,
  child: PropTypes.number,
  infant: PropTypes.number,
  totalGuests: PropTypes.number,
  maxGuests: PropTypes.number,

};
Dropdown.defaultProps = {
  clicked: false,
  btnClick: () => null,
  close: () => null,
  adult: 0,
  child: 0,
  infant: 0,
  totalGuests: 0,
  maxGuests: 0,
};
