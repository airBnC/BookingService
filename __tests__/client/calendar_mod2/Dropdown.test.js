import Adapter from 'enzyme-adapter-react-16';
import React from 'react';
import 'jest-styled-components';
import { configure, mount } from 'enzyme';
import renderer from 'react-test-renderer';
import { DropdownHolder } from '../../../client/src/ModStylings';
import Calendar from '../../../client/src/calendar_mod2/Calendar';
import Dropdown from '../../../client/src/calendar_mod2/Dropdown';

configure({ adapter: new Adapter() });

const props = {
  clicked: 'checkIn',
};

describe('Holder Component', () => {
  it('is the main flex component', () => {
    const tree = renderer
      .create(<DropdownHolder> things inside </DropdownHolder>)
      .toJSON();
    expect(tree).toHaveStyleRule('display', 'flex');
    expect(tree).toMatchSnapshot();
  });
});
describe('Dropdown Component', () => {
  it('renders the right amount of children', () => {
    const dropdown = mount(<Dropdown {...props} />);
    expect(dropdown.find(Calendar)).toHaveLength(1);
    expect(dropdown).toMatchSnapshot();
  });
  it('does not render if not clicked', () => {
    const dropdown = mount(<Dropdown clicked="" />);
    expect(dropdown).toMatchSnapshot();
  });
});
