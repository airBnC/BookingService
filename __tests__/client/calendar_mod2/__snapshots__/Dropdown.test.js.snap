// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dropdown Component does not render if not clicked 1`] = `
<Dropdown
  arrowClick={[Function]}
  calendarChange={[Function]}
  clearDates={[Function]}
  clicked=""
  currDate={null}
  dates={Array []}
  endDate={null}
  minStay={0}
  month={null}
  setDate={[Function]}
  startDate={null}
  year={null}
/>
`;

exports[`Dropdown Component renders the right amount of children 1`] = `
.c1 {
  position: absolute;
  z-index: 2;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  width: 286px;
  border: 1px solid rgb(172,172,172);
  border-radius: 1%;
  padding: 20px;
  background-color: white;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

.c10 {
  margin-top: 8px;
  border: none;
  -webkit-align-self: flex-end;
  -ms-flex-item-align: end;
  align-self: flex-end;
  font-size: 14px;
  color: #007D8C;
  outline: none;
}

.c10:hover {
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.c10:active {
  color: #F16664;
}

.c6 {
  margin-top: 20px;
  border-style: hidden;
  border-collapse: collapse;
  font-weight: bold;
}

.c7 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
}

.c8 {
  min-width: 38px;
  min-height: 25px;
  font-size: 13px;
  text-align: center;
}

.c3 {
  border-radius: 10%;
  content: url(process:function process(src,filename) { return 'module.exports = ' + JSON.stringify(path.basename(filename)) + ';';
}

;);height:20px;border:1px solid rgb(172,172,172);border-radius:10%;padding:3px;-webkit-transform:rotate(0.5turn);-ms-transform:rotate(0.5turn);transform:rotate(0.5turn);} .c3:active {
  background-color: #79CCCD;
}

.c3:hover {
  cursor: pointer;
}

.c4 {
  border-radius: 10%;
  content: url(process:function process(src,filename) { return 'module.exports = ' + JSON.stringify(path.basename(filename)) + ';';
}

;);height:20px;border:1px solid rgb(172,172,172);border-radius:10%;padding:3px;} .c4:active {
  background-color: #79CCCD;
}

.c4:hover {
  cursor: pointer;
}

.c0 {
  border-color: #ffffff #ffffff #acacac #ffffff;
  border-style: solid;
  border-width: 0px 8px 12px 8px;
  height: 5px;
  width: 0px;
  margin-left: 20px;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  margin-top: 10px;
  font-size: 18px;
  font-weight: bold;
}

.c5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c9 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  margin-top: 12px;
  font-size: 12px;
}

<Dropdown
  arrowClick={[Function]}
  calendarChange={[Function]}
  clearDates={[Function]}
  clicked="checkIn"
  currDate={null}
  dates={Array []}
  endDate={null}
  minStay={0}
  month={null}
  setDate={[Function]}
  startDate={null}
  year={null}
>
  <div>
    <DropdownStyling__Triangle
      left="checkIn"
    >
      <div
        className="c0"
      />
    </DropdownStyling__Triangle>
    <ModStylings__DropdownHolder>
      <div
        className="c1"
      >
        <DropdownStyling__SectionOne>
          <div
            className="c2"
          >
            <DropdownStyling__Arrow
              left={true}
              onClick={[Function]}
            >
              <div
                className="c3"
                onClick={[Function]}
              />
            </DropdownStyling__Arrow>
            <div>
               
               
               
            </div>
            <DropdownStyling__Arrow
              onClick={[Function]}
            >
              <div
                className="c4"
                onClick={[Function]}
              />
            </DropdownStyling__Arrow>
          </div>
        </DropdownStyling__SectionOne>
        <DropdownStyling__SectionTwo>
          <div
            className="c5"
          >
            <Calendar
              arrowClick={[Function]}
              calendarChange={[Function]}
              clearDates={[Function]}
              clicked="checkIn"
              currDate={null}
              dates={Array []}
              endDate={null}
              minStay={0}
              month={null}
              setDate={[Function]}
              startDate={null}
              year={null}
            >
              <CalendarStyling__Table>
                <table
                  className="c6"
                >
                  <thead>
                    <CalendarStyling__Tr>
                      <tr
                        className="c7"
                      >
                        <CalendarStyling__Td
                          key="Su"
                        >
                          <td
                            className="c8"
                          >
                            <div>
                               
                              Su
                               
                            </div>
                          </td>
                        </CalendarStyling__Td>
                        <CalendarStyling__Td
                          key="Mo"
                        >
                          <td
                            className="c8"
                          >
                            <div>
                               
                              Mo
                               
                            </div>
                          </td>
                        </CalendarStyling__Td>
                        <CalendarStyling__Td
                          key="Tu"
                        >
                          <td
                            className="c8"
                          >
                            <div>
                               
                              Tu
                               
                            </div>
                          </td>
                        </CalendarStyling__Td>
                        <CalendarStyling__Td
                          key="We"
                        >
                          <td
                            className="c8"
                          >
                            <div>
                               
                              We
                               
                            </div>
                          </td>
                        </CalendarStyling__Td>
                        <CalendarStyling__Td
                          key="Th"
                        >
                          <td
                            className="c8"
                          >
                            <div>
                               
                              Th
                               
                            </div>
                          </td>
                        </CalendarStyling__Td>
                        <CalendarStyling__Td
                          key="Fr"
                        >
                          <td
                            className="c8"
                          >
                            <div>
                               
                              Fr
                               
                            </div>
                          </td>
                        </CalendarStyling__Td>
                        <CalendarStyling__Td
                          key="Sa"
                        >
                          <td
                            className="c8"
                          >
                            <div>
                               
                              Sa
                               
                            </div>
                          </td>
                        </CalendarStyling__Td>
                      </tr>
                    </CalendarStyling__Tr>
                  </thead>
                  <tbody>
                    <CalendarStyling__Tr>
                      <tr
                        className="c7"
                      >
                        <td />
                      </tr>
                    </CalendarStyling__Tr>
                  </tbody>
                </table>
              </CalendarStyling__Table>
            </Calendar>
          </div>
        </DropdownStyling__SectionTwo>
        <DropdownStyling__SectionThree>
          <div
            className="c9"
          >
            <div>
               
              0
               day minimum. 
            </div>
            <div>
               Updated 2 days ago 
            </div>
          </div>
        </DropdownStyling__SectionThree>
        <ModStylings__Close
          onClick={[Function]}
        >
          <button
            className="c10"
            onClick={[Function]}
          >
             Clear Dates 
          </button>
        </ModStylings__Close>
      </div>
    </ModStylings__DropdownHolder>
  </div>
</Dropdown>
`;

exports[`Holder Component is the main flex component 1`] = `
.c0 {
  position: absolute;
  z-index: 2;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  width: 286px;
  border: 1px solid rgb(172,172,172);
  border-radius: 1%;
  padding: 20px;
  background-color: white;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

<div
  className="c0"
>
   things inside 
</div>
`;
